src/acme/ui/CustomerListUI.java

```diff
@@ -0,0 +1,353 @@
+import javax.swing.table.TableCellEditor;
+import javax.swing.table.TableCellRenderer;
+import javax.swing.table.TableColumn;
+
+
+
+import acme.pd.User;
+import acme.pd.Customer;
+import acme.pd.MapIntersection;
+import acme.pd.Road;
+
+public class CustomerListUI extends AcmeBaseJPanel {
+  Customer [] customersArray;
+  
+  public CustomerListUI() {
+    super();
+  }
+  
+  public void buildPanel() {
+  
+    setLayout(null);
+    
+    Color lightBlue = new Color(93, 184, 202);
+    
+    //title label
+    JLabel lblCustomers = new JLabel("Customers");
+    lblCustomers.setFont(new Font("Lucida Grande", Font.PLAIN, 24));
+    lblCustomers.setBounds(20, 20, 200, 27);
+    add(lblCustomers);
+    
+    //my test data
+    HashMap<UUID, Customer> testData = getCompany().getCustomer();
+    Collection<Customer> values = testData.values();
+    ArrayList<Customer> listOfValues = new ArrayList<Customer>(values);
+    Set<UUID> keySet = testData.keySet();
+    ArrayList<UUID> listOfKeys = new ArrayList<UUID>(keySet);
+    customersArray = listOfValues.toArray(new Customer[listOfValues.size()]);
+    Object[][] objects = getObjectsForTable(customersArray);
+    String[] cols = new String [] {
+        "Number", "Name", "Location", "Active", "Edit"
+    };
+    
+    
+
+    //scroll and table. I made this in netbeans, so it looks a little different
+    JScrollPane jScrollPane1;
+    JTable jTable1 = new JTable();
+
+    //set up table
+    jTable1.setBorder(javax.swing.BorderFactory.createEtchedBorder());
+    jTable1.setFont(new java.awt.Font("Lucida Grande", 0, 16)); 
+    jTable1.setModel(new javax.swing.table.DefaultTableModel(objects, cols) {
+        Class[] types = new Class [] {
+            java.lang.String.class, java.lang.String.class, java.lang.String.class, java.lang.Boolean.class, javax.swing.JButton.class
+        };
+        boolean[] canEdit = new boolean [] {
+            //need last one true so can click button
+            false, false, false, false, true
+        };
+
+        public Class getColumnClass(int columnIndex) {
+            return types [columnIndex];
+        }
+
+        public boolean isCellEditable(int rowIndex, int columnIndex) {
+            return canEdit [columnIndex];
+        }
+    });
+    
+    //needed for buttons in table
+    TableColumn editCol = jTable1.getColumnModel().getColumn(4);
+    JButton editButton = new JButton("edit");
+    editCol.setCellRenderer(new ButtonRenderer());;
+    editCol.setCellEditor(new ButtonEditor(new JTextField()));
+  
+    //more table properties
+    jTable1.setColumnSelectionAllowed(false);
+    jTable1.getTableHeader().setFont(new Font("Lucida Grande", 0, 18)); 
+    jTable1.setRowHeight(24);
+    jTable1.setRowMargin(2);
+    
+    jTable1.setSelectionBackground(new Color(93, 184, 202));
+    jTable1.setSelectionForeground(new java.awt.Color(0, 0, 0));
+    jTable1.setShowGrid(true);
+    jTable1.setPreferredScrollableViewportSize(jTable1.getPreferredSize());
```

Comment:
I have to resize the JFrame in order to see all of the columns. Can you resize the table or make it autoResize so it all fits in the JFrame?

Reply:
I fixed this.